src/tokenizer/mod.rs

```diff
@@ -1397,12 +1402,118 @@ impl<Sink: TokenSink> Tokenizer<Sink> {
     }
 }
 
+
+
+
+
 #[cfg(test)]
 #[allow(non_snake_case)]
 mod test {
     use super::option_push; // private items
     use tendril::{StrTendril, SliceExt};
 
+    use std::io::{self};
+    use super::{TokenSink, Tokenizer, TokenizerOpts, TokenSinkResult};
+
+    pub use super::interface::{Token, DoctypeToken, TagToken, CommentToken};
+    pub use super::interface::{CharacterTokens, NullCharacterToken, EOFToken, ParseError};
+    pub use super::interface::{Doctype, Attribute, TagKind, StartTag, EndTag, Tag};
+
+    use super::buffer_queue::{BufferQueue};
+    use std::mem::replace;
+    use std::borrow::Cow::{self, Borrowed};
+
+    // TokenMatch implements the TokenSink trait. It is used for testing to see 
+    // if current_line is being updated when process_token is called. The lines
+    // vector is a collection of the line numbers that each token is on.
+    struct TokenMatch {
+        tokens: Vec<Token>,
+        current_str: StrTendril,
+        exact_errors: bool,
+        lines: Vec<u64>,
```

Comment:
Let's make this `Vec<(Token, u64)>`, so we can verify that the tokens produced are what we expect.